### ISSUE TITLE
[SUGGESTION] Some additions I think would be beneficial.

### ISSUE THREAD

**Issue by Clark919** [kirpichyov/LC_EnemiesScanner#7]
Love the mod, it's super cool. Some suggestions I would recommend considering:

- Config to split the device into four separate modes that the user can specify to select what they want to use for their lobbies:
   - Simple: Only displays the monster counter and beeps when in proximity of one.
   - List: Functions exactly how it does currently.
   - Radar: Provides a radar UI on the screen that shows close enemies as a blip. Alien Isolation's radar is a sort of good reference for what I'm talking about here.
   - Combined: Has both the functionalities of the list and radar settings, to show what is near and exactly where it is in relation to the user.
- Option to blacklist/whitelist certain entities. Turrets in particular being pinged is quite annoying and renders this device a bit useless when it's constantly pinging.
- Option to change the aliases of entities. Have a section generated in the config each time the player loads into a game so the game registers all entities (including modded). This would allow players a lot of customisation of what entities are referred to as and they could even obscure some entities in particular this way if they want dangerous/unknown entities to appear. Example of some lines in the config would be like:
```
Alias_Flowerman = "Bracken"
Alias_Centipede = "Snare Flea"
Alias_Boomba = "Boomba"
```
- Adding onto this last suggestion, a way to sync entities marked with the ECHO scanner with this mod would be very cool. A Jester may show up as "Unknown" on the scanner until you mark it with a scan for the first time (getting the default added to bestiary message) at which point it would then show as Jester. Obviously you'd have to add a blacklist for this also as some monsters physically cannot be scanned (Such as the Masked and some modded monsters).

**Comment by Clark919** [comment]
One other thing I'd add on is maybe reducing the model size and moving it up a bit? It's super hard to see the heat meter sometimes with that setting on is all, and the device can be visually obtrusive (I know that might be a balance choice which I'm inclined to agree with since it's a strong device).

**Comment by kirpichyov** [comment]
Hi, @Clark919! Thanks for such great and detailed feedback. The suggestions and issues listed by you look pretty justified, helpful, and interesting to me. I am not sure if I will be able to implement everything of that or implement it in one update at once. But I will try. Appreciate your feedback!

**Comment by kirpichyov** [comment]
Hi, @Clark919! Thanks for such great and detailed feedback. The suggestions and issues listed by you look pretty justified, helpful, and interesting to me. I am not sure if I will be able to implement everything of that or implement it in one update at once. But I will try. Appreciate your feedback!

**Comment by Clark919** [comment]
> Hi, @Clark919! Thanks for such great and detailed feedback. The suggestions and issues listed by you look pretty justified, helpful, and interesting to me. I am not sure if I will be able to implement everything of that or implement it in one update at once. But I will try. Appreciate your feedback!

Glad you're looking into it! I'm adding your mod to [TheMostLethalCompany](https://thunderstore.io/c/lethal-company/p/TeamClark/TheMostLethalCompany/) today, it's one of the largest in the community and I'm looking forward to seeing if people like it as much as myself :)

**Comment by kirpichyov** [comment]
Hi! After some prototyping and investigation, I have decided to close the issue. 
I was able to implement the next features:
- Option to blacklist certain entities
- Option to change the aliases of entities
- Reducing the model size and moving it up a bit
- Moving the heat indicator to the top

Some of the features I will not implement since it will be hard to maintain and develop the mod if introduced. Also, some technical limitations require a lot of time from my side to bypass it. There is a list of these features:
- Config to split the device into four separate modes (introduce radar)
- Sync entities marked with the ECHO scanner 

Thanks for your feedback!
